backend/mongo/store.go

```diff
@@ -304,6 +317,15 @@ func (s *EventStore) Insert(ctx context.Context, events ...event.Event) error {
 			return fmt.Errorf("update state: %w", err)
 		}
 
+		for _, fn := range s.txEventHandlers {
```

Comment:
We probably want to this if we are inside a transaction, otherwise it doesn't make sense, so probably we need to check if a transaction is in place, or force a transaction start (`s.transactions = true` at the beginning) if there is any txEventHandlers

Reply:
this will only be non-empty if `s.transactions` is true because the `WithTxEventHandler()` will set it to true

Reply:
but I might need to check if `s.transactions` are not being set to false using the `Transactions(tx bool)` option

Reply:
maybe you should the other way around. If you specify transaction to true when defining a txEventHandler, it is because we want a transaction, so we set transaction to true. And as you said if we disabled the transaction, we should check if is still valid when we run

Reply:
now panicing if transactions are set to false after a handler has been registered